### PR TITLE
Java alb

### PR DESCRIPTION
**Pull request by danizettle**
adding a common package to register and de-register a dropwizard application in aws target group

ping @tbaylis , @fredrikbackstrom

### CODE REVIEW

**Comment by tbaylis** [comment]
Seems ok, but maybe some documentation describing how to use the package would be nice?

**Comment by adamvoncorswant** [comment]
Also, nitpicking: please use spaces instead of tabs, and make sure each file has a newline at EOF.

File: izettle-java-alb/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# izettle-java-alb
+
+Module to use together with Dropwizard. Should be registered in the Application class 
+and will be called on start and stop of the application server, either during deployment or a 
+restart of the instance itself. 
+
+Authentication roles must be provided by IAM.
```
Comment (tbaylis) [inline izettle-java-alb/README.md]: Don't we require a `AmazonElasticLoadBalancingClient`, not caring about where it gets its credentials from?

Comment (danizettle) [inline izettle-java-alb/README.md]: True dat

File: izettle-java-alb/src/main/java/com/izettle/alb/Condition.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+public interface Condition {
+
+    /**
+     * Await a precondition is met for a specified max duration.
+     * @param executorServiceSupplier Supplier of a executor service
+     * @param notificationsConsumer A consumer that can take string messages from the precondition
+     * @return
+     */
+    CompletableFuture<Boolean> check(
+        Supplier<ExecutorService> executorServiceSupplier,
```
Comment (xiaodong-izettle) [inline izettle-java-alb/src/main/java/com/izettle/alb/Condition.java]: Maybe just a plain `ExecutorService  executorService` is enough?

File: izettle-java-alb/src/main/java/com/izettle/alb/Condition.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+/**
+ * Interface for classes that implement a condition check.
+ */
+public interface Condition {
```
Comment (xiaodong-izettle) [inline izettle-java-alb/src/main/java/com/izettle/alb/Condition.java]: Some nitpick, maybe add `@FunctionalInterface` for this interface, if it's intended to use `lambda` syntax... :)

File: izettle-java-alb/src/main/java/com/izettle/alb/ALBAvailableInstancesCondition.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Override
+    public CompletableFuture<Boolean> check(
+        Supplier<ExecutorService> executorServiceProvider,
+        Consumer<String> notificationsConsumer
+    ) {
+
+        return CompletableFuture.supplyAsync(
+            () -> {
+                while (true) {
```
Comment (danizettle) [inline izettle-java-alb/src/main/java/com/izettle/alb/ALBAvailableInstancesCondition.java]: We will look at the ansible script while deploying so we will see if something is up. Only alternative is to fail the startup.

**Pull request merged**